src/lib/y2users/user.rb

```diff
@@ -113,15 +113,21 @@ def primary_group
       config.groups.find { |g| g.gid == gid }
     end
 
-    # Groups where the user is included. It also contains the primary group.
+    # Groups where the user is included. It also contains the primary group unless set otherwise.
     #
     # The user must be attached to a config in order to find its groups.
     #
+    # @param with_primary [Boolean] if primary group should be included
     # @return [Array<Group>]
-    def groups
+    def groups(with_primary: true)
       return [] unless attached?
 
-      config.groups.select { |g| g.users.include?(self) }
+      res = config.groups.select { |g| g.users.include?(self) }
+
+      primary_group_cache = primary_group
+      res.delete_if { |g| g == primary_group_cache } unless with_primary
+
```

Comment:
NP: Maybe this is more straightforward?

~~~
groups = config.groups.select { |g| g.users.include?(self) }
groups.reject { |g| g.gid == gid } unless with_primary
~~~

Reply:
sadly not. As it is nil if with_primary is false. Try in IRB
```
a = [0,1]
=> [0, 1]
irb(main):003:0> a.reject(&:even?) if false
=> nil
```

Reply:
Yes, sorry, what I wanted to propose is to use #gid directly:

~~~
groups = config.groups.select { |g| g.users.include?(self) }
groups.reject! { |g| g.gid == gid } unless with_primary
groups
~~~

Reply:
ah, it makes sense then